public/app/features/dashboard/services/TimeSrv.ts

```diff
@@ -77,13 +78,14 @@ export class TimeSrv {
     );
 
     if (range.to.isBefore(range.from)) {
-      this.setTime(
-        {
-          from: range.raw.to,
-          to: range.raw.from,
-        },
-        false
-      );
+      const appEvents = getAppEvents();
+      appEvents.publish({
+        type: AppEvents.alertWarning.name,
+        payload: [
+          'Invalid time range',
+          'This dashboard may not show any data as the time range is invalid: `to` value is before `from`.',
```

Comment:
My only comment is that I think the warning should say something similar to the error

```suggestion
          'This dashboard may not show any data as the time range is invalid: "From" value is after "To".',
```